tooling/debugger/tests/debug.rs

```diff
@@ -49,4 +49,115 @@ mod tests {
+
+        let timeout_seconds = 30;
+        let mut dbg_session =
+            spawn_bash(Some(timeout_seconds * 1000)).expect("Could not start bash session");
+
+        let test_program_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("../../test_programs/execution_success/regression_7195");
+        let test_program_dir = test_program_path.display();
+
+        // Start debugger and test that it loads for the given program.
+        dbg_session
+            .execute(
+                &format!(
+                    "{nargo_bin} debug --raw-source-printing true --program-dir {test_program_dir} --force-brillig --expression-width 3"
+                ),
+                ".*\\Starting debugger.*",
+            )
+            .expect("Could not start debugger");
+
+        let num_steps = 16;
+        for _ in 1..=num_steps {
+            // While running the debugger, issue a "next" cmd,
+            // which should run to the program to the next source line given
+            // we haven't set any breakpoints.
+            // ">" is the debugger's prompt, so finding one
+            // after running "next" indicates that the
+            // debugger has not panicked for this step.
+            dbg_session
+                .send_line("next")
+                .expect("Debugger panicked while attempting to step through program.");
+            dbg_session
+                .exp_string(">")
+                .expect("Failed while waiting for debugger to step through program.");
+        }
+
+        let mut lines = vec![];
+        while let Ok(line) = dbg_session.read_line() {
+            if !(line.starts_with(">next") || line.starts_with("At ") || line.starts_with("...")) {
+                lines.push(line);
+            }
+        }
+
+        let lines_expected_to_contain: Vec<&str> = vec![
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "}",
+            "> next",
+            "    let x = unsafe { baz(x) };",
+            "> next",
+            "    foo(x);",
+            "fn foo(x: Field) {",
+            "> next",
+            "    foo(x);",
+            "fn foo(x: Field) {",
+            "> next",
+            "    foo(x);",
+            "    let y = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
+            "> next",
+            "    foo(x);",
+            "    let y = unsafe { baz(x) };",
+            "unconstrained fn baz(x: Field) -> Field {",
+            "> next",
+            "    foo(x);",
+            "    let y = unsafe { baz(x) };",
+            "}",
+            "> next",
+            "    foo(x);",
+            "    let y = unsafe { baz(x) };",
```

Comment:
Just curious: why do we have seem to have everything appear twice?

Reply:
It's skipping the highlighting moving from e.g. all of `foo(x)` to just `x`.